### PR TITLE
Department  Access and Membership

### PR DESCRIPTION
**Pull request by protich**
This PR addresses a bug in department membership check and also adds a few enhancements.

__Todo__
- [ ] Rebase PR after #2996 is merged to drop use of `all()-asArray()`.

### CODE REVIEW

File: include/class.dept.php [hunk 1]
```diff
@@ -205,6 +200,42 @@ function getAvailableMembers() {
...
+
+    function getExtendedMembers() {
+
+        if (!isset($this->_exended_members)) {
+            // We need a query set so we can sort the names
+            $members = StaffDeptAccess::objects();
+            $members->filter(array('dept_id' => $this->getId()));
+            $members = Staff::nsort($members, 'staff__');
+            $extended = array();
+            foreach($members->all()->asArray() as $member) {
```
Comment (greezybacon) [inline include/class.dept.php]: You can iterate with `->all()` always. This issue was that it used to return an array and it was changed to return an iterator. #2996 converts it back to an array. Either way, asArray() wouldn't be necessary to iterate

File: include/class.dept.php [hunk 2]
```diff
@@ -205,6 +200,42 @@ function getAvailableMembers() {
...
+                if (!$member->staff) continue;
+                // Annoted the staff model with alerts and role
+                $extended[] = new AnnotatedModel($member->staff, array(
```
Comment (greezybacon) [inline include/class.dept.php]: I worked on a variation to the InstrumentedList in the custom queue feature to automate this trick. Perhaps we come up with a nice way to describe it in the meta data without duplicating code

Comment (protich) [inline include/class.dept.php]: Perhaps expanded use of broker?

Comment (greezybacon) [inline include/class.dept.php]: Yeah, it was a special broker subclass

**Pull request merged**